src/System.IO.FileSystem/tests/Enumeration/RemovedDirectoryTests.netcoreapp.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Enumeration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests.Enumeration
+{
+    public class RemovedDirectoryTests : FileSystemTest
+    {
+        private class DirectoryFinishedEnumerator : FileSystemEnumerator<string>
+        {
+            public DirectoryFinishedEnumerator(string directory, EnumerationOptions options)
+                : base(directory, options)
+            {
+            }
+
+            protected override string TransformEntry(ref FileSystemEntry entry)
+            {
+                return entry.FileName.ToString();
+            }
+
+            public delegate void DirectoryFinishedDelegate(ReadOnlySpan<char> directory);
+
+            public DirectoryFinishedDelegate DirectoryFinishedAction { get; set; }
+
+            protected override void OnDirectoryFinished(ReadOnlySpan<char> directory)
```

Comment:
When is this invoked?  It doesn't seem like it would exercise the case called out in https://github.com/dotnet/corefx/issues/28819#issuecomment-378820810.

Reply:
Oh, nevermind, I see, you're deleting the sub-directory when the parent directory has been enumerated, so you're then about to enumerate the subdirectory.

Reply:
Whenever you get to the end of a directory. So we've put the subdir on the stack by that point.